themes/fishy.zsh-theme

```diff
@@ -1,13 +1,13 @@
 # ZSH Theme emulating the Fish shell's default prompt.
 
 _fishy_collapsed_wd() {
-  echo $(pwd | perl -pe "
+  echo $(pwd | perl -pe '
    BEGIN {
-      binmode STDIN,  ':encoding(UTF-8)';
-      binmode STDOUT, ':encoding(UTF-8)';
-   }; s|^$HOME|~|g; s|/([^/])[^/]*(?=/)|/\$1|g
-")
-} 
+      binmode STDIN,  ":encoding(UTF-8)";
+      binmode STDOUT, ":encoding(UTF-8)";
+   }; s|^$HOME|~|g; s|/([^/.])[^/]*(?=/)|/$1|g; s|/\.([^/])[^/]*(?=/)|/.$1|g;
```

Comment:
> For example, here the $1 is not supposed to be interpolated by the shell (notice the slash escaping the dollar sign before it in the double-quotes version)

Oh right, I hadn't noticed that (you're referring to [this line](https://github.com/robbyrussell/oh-my-zsh/pull/4667#discussion-diff-47481372L8), of course). I agree with your solution then.